models/models/search.py

```diff
@@ -33,9 +33,16 @@ class SampleSearchResponseData(SearchResponseData):
     sample_external_ids: list[str]
 
 
+class ErrorSearchResponseData(SearchResponseData):
+    """Error search response data"""
+
+    error: str
+
+
 class SearchResponse(SMBase):
     """Response class for each search result"""
 
     type: SearchResponseType
     title: str
     data: SampleSearchResponseData | ParticipantSearchResponseData | FamilySearchResponseData
+    error: ErrorSearchResponseData | None
```

Comment:
I think providing a default here means you don't have to specify it when constructing (additionally allowing it to be None):

```suggestion
    error: ErrorSearchResponseData | None = None
```